### PR TITLE
BePhoneBook [recipe]

### PR DESCRIPTION
**Pull request by bach5000**
The patch is just there to allow the recipe to build. I've already made a PR on the upstream source:
https://github.com/HaikuArchives/BePhoneBook/pull/1
Will remove patch once the upstream PR is merged.

Also, BePhoneBook uses this license: https://github.com/HaikuArchives/BePhoneBook/blob/69d6a5ada4fa76b0e900face8f6eef1d7ac48119/Source/CLVColumnLabelView.cpp#L2

I'm pretty sure this is a custom license, but what do I put for `LICENSE`?

### CODE REVIEW

File: haiku-apps/bephonebook/bephonebook-1.2.recipe [hunk 1]
```diff
@@ -0,0 +1,48 @@
+SUMMARY="A phone book manager"
+DESCRIPTION="BePhoneBook is a colorful phonebook made to help you stay \
+organized with your family's or friends' phone numbers."
+HOMEPAGE="https://github.com/HaikuArchives/BePhoneBook"
+COPYRIGHT="1999 Michael Melloff \
+	Fabien Fulhaber"
+LICENSE="custom"
+REVISION="1"
+srcGitRev="ca884c1af8bee25d71389df85001757bef4a3a6a"
```
Comment (fbrosson) [inline haiku-apps/bephonebook/bephonebook-1.2.recipe]:
```diff
-srcGitRev="ca884c1af8bee25d71389df85001757bef4a3a6a"
+srcGitRev="69d6a5ada4fa76b0e900face8f6eef1d7ac48119"
```
The srcGitRev you have right now is the one from `haiku-apps/2dphysicsdemo/2dphysicsdemo-22.5.98.recipe` :smile:

**Comment by scottmc** [comment]
The license you listed is for LibSanta.  You might check the program's use of LibSanta and see if they modified it, or used it as it was.  If un changed we might be able to remove it from here and just add it as a build dependency in the recipe.

File: haiku-apps/bephonebook/bephonebook-1.2.recipe (hunk 1 above)
Comment (bach5000) [reply haiku-apps/bephonebook/bephonebook-1.2.recipe]: @fbrosson Oops! Must have forgotten to change it.

File: haiku-apps/bephonebook/bephonebook-1.2.recipe [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+LICENSE="BePhoneBook"
+REVISION="1"
+srcGitRev="69d6a5ada4fa76b0e900face8f6eef1d7ac48119"
+SOURCE_URI="https://github.com/HaikuArchives/bephonebook/archive/$srcGitRev.tar.gz"
+CHECKSUM_SHA256="31a4b3c72905a789746509801ee667ebb1ef3b7dbd57c63623b83fc4b064ccc6"
+SOURCE_FILENAME="BePhoneBook-$portVersion-$srcGitRev.tar.gz"
+SOURCE_DIR="BePhoneBook-$srcGitRev"
+PATCHES="bephonebook-$portVersion.patchset"
+
+ARCHITECTURES="?x86_gcc2 x86_64"
```
Comment (fbrosson) [inline haiku-apps/bephonebook/bephonebook-1.2.recipe]:
```diff
-ARCHITECTURES="?x86_gcc2 x86_64"
+ARCHITECTURES="x86_gcc2 x86_64"
```
(The app builds fine with gcc2 and seems to also work the same way it does on x86_64.)

File: haiku-apps/bephonebook/bephonebook-1.2.recipe [hunk 3]
```diff
@@ -0,0 +1,48 @@
+SUMMARY="A phone book manager"
+DESCRIPTION="BePhoneBook is a colorful phonebook made to help you stay \
+organized with your family's or friends' phone numbers."
+HOMEPAGE="https://github.com/HaikuArchives/BePhoneBook"
+COPYRIGHT="1999 Michael Melloff \
+	Fabien Fulhaber"
+LICENSE="BePhoneBook"
+REVISION="1"
+srcGitRev="69d6a5ada4fa76b0e900face8f6eef1d7ac48119"
```
Comment (fbrosson) [inline haiku-apps/bephonebook/bephonebook-1.2.recipe]:
```diff
-srcGitRev="69d6a5ada4fa76b0e900face8f6eef1d7ac48119"
+rcGitRev="666d1cf3ea715f20455576664823c8ff85b4eec0
```
(Don't forget to drop the patchset and update `CHECKSUM_SHA256`)

Oops, you also need to add `haiku-apps/bephonebook/licenses/BePhoneBook`.

File: haiku-apps/bephonebook/bephonebook-1.2.recipe [hunk 4]
```diff
@@ -0,0 +1,48 @@
+SUMMARY="A phone book manager"
+DESCRIPTION="BePhoneBook is a colorful phonebook made to help you stay \
+organized with your family's or friends' phone numbers."
+HOMEPAGE="https://github.com/HaikuArchives/BePhoneBook"
+COPYRIGHT="1999 Michael Melloff \
+	Fabien Fulhaber"
+LICENSE="BePhoneBook"
```
Comment (alaviss) [inline haiku-apps/bephonebook/bephonebook-1.2.recipe]: Please add the license text to `licenses/libsanta` relative to the recipe and change `LICENSES` to `libsanta`.

While a lot of the code uses the linked license, the main files are **not** covered by this license, so you'd need to add `Unknown` there.

**Comment by bach5000** [comment]
@alaviss @fbrosson:
Sorry for the late reply! Just saw this now.
For the license folder, do I add a license file called "BePhoneBook" or "libsanta"?
And for the file's contents, do I add the license from here? https://github.com/HaikuArchives/BePhoneBook/blob/69d6a5ada4fa76b0e900face8f6eef1d7ac48119/Source/CLVColumnLabelView.cpp#L2

**Comment by alaviss** [comment]
Yep, call it "libsanta"

**Comment by bach5000** [comment]
@alaviss And add that license snippet into the file, right?

**Comment by bach5000** [comment]
@alaviss Does this look correct?

File: haiku-apps/bephonebook/bephonebook-1.2.recipe [hunk 5]
```diff
@@ -0,0 +1,47 @@
+SUMMARY="A phone book manager"
+DESCRIPTION="BePhoneBook is a colorful phonebook made to help you stay \
+organized with your family's or friends' phone numbers."
+HOMEPAGE="https://github.com/HaikuArchives/BePhoneBook"
+COPYRIGHT="1999 Michael Melloff \
```
Comment (alaviss) [inline haiku-apps/bephonebook/bephonebook-1.2.recipe]: Why is there a `\` here?

File: haiku-apps/bephonebook/bephonebook-1.2.recipe [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+	Fabien Fulhaber"
+LICENSE="libsanta"
```
Comment (alaviss) [inline haiku-apps/bephonebook/bephonebook-1.2.recipe]:
```suggestion
LICENSE="libsanta
	Unknown"
```

As [mentioned](https://github.com/haikuports/haikuports/pull/3354#discussion_r232650490), the main files don't have any license associated with, so we'd still need `Unknown` here.

File: haiku-apps/bephonebook/bephonebook-1.2.recipe (hunk 5 above)
Comment (bach5000) [reply haiku-apps/bephonebook/bephonebook-1.2.recipe]: @alaviss Are escape characters not needed?

Comment (alaviss) [reply haiku-apps/bephonebook/bephonebook-1.2.recipe]: Of course it's allowed, but why would you need it? And is there any copyright year for the second entry?

File: haiku-apps/bephonebook/bephonebook-1.2.recipe [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+	"
+BUILD_PREREQUIRES="
+	makefile_engine
+	cmd:g++
+	cmd:make
+	"
+
+BUILD()
+{
+	make OBJ_DIR=objects
```
Comment (alaviss) [inline haiku-apps/bephonebook/bephonebook-1.2.recipe]:
```suggestion
	make $jobArgs OBJ_DIR=objects
```

Because why not?

**Comment by bach5000** [comment]
@alaviss Oops. I was suppose to copy "about.jpg" to the same directory as the app.
Let me change that :)

**Comment by scottmc** [comment]
For the about box image, I would turn it into an rdef and then make it part of the binary, then there's no stay .jpg to worry about.  This is how it was done in BeVexed and HexVexed.  This isn't a requirement, just suggesting it.

**Comment by Begasus** [comment]
I think this can be merged? The "Unknown" license issue will remain so haikuporter wont' pick it up in the buildbot, anyone in need for it can build it themselves then.

**Comment by korli** [comment]
> For the about box image, I would turn it into an rdef and then make it part of the binary, then there's no stay .jpg to worry about. This is how it was done in BeVexed and HexVexed. This isn't a requirement, just suggesting it.

@scottmc do you want to create an issue with this?

**Pull request merged**